.github/workflows/build.yml

```diff
@@ -0,0 +1,266 @@
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+             ${{ runner.os }}-buildx-
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PERSONAL_ACCESS_TOKEN }}
+
+      - name: Build
+        uses: docker/build-push-action@v2
+        with:
+          push: ${{ github.ref == 'refs/heads/master' }}
+          builder: ${{ steps.buildx.outputs.name }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+          tags: |
+                 ${{env.DOCKERHUB_REPOSITORY}}:sha-${{ steps.vars.outputs.sha_short }}
+          build-args: GIT_COMMIT_SHA=${{ steps.vars.outputs.sha_short }}
+
+      - name: Slack Notification
+        if: failure()
+        uses: rtCamp/action-slack-notify@master
+        env:
+           SLACK_CHANNEL: getintoteaching_tech
+           SLACK_COLOR: '#3278BD'
+           SLACK_ICON: https://github.com/rtCamp.png?size=48
+           SLACK_MESSAGE: 'The pipeline has failed to build the API image'
+           SLACK_TITLE: 'Failure to Build API '
+           SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+           SLACK_FOOTER: ${{ env.APPLICATION }}
+
+  development:
+    name: Development Deployment
+    needs: build
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    outputs:
+      release_tag: ${{steps.tag_version.outputs.tag}}
+      release_sha: ${{steps.tag_version.outputs.sha}}
+    steps:
+       - name: Trigger Development Deployment
+         uses: benc-uk/workflow-dispatch@v1.1
+         with:
+           workflow: Deploy to PaaS
+           token: ${{ secrets.ACTIONS_API_ACCESS_TOKEN }}
+           inputs: '{"environment": "Development", "sha": "${{ github.sha }}"}'
+           ref: ${{github.ref}}
+
+       - name: Wait for Deployment to Development
+         uses: fountainhead/action-wait-for-check@v1.0.0
+         id: wait-for-deploy
+         with:
+           token: ${{ secrets.ACTIONS_API_ACCESS_TOKEN}}
+           checkName: Deploy Development
+           ref: ${{github.ref}}
+
+       - name: Check if Development Deployment has returned with a failure
+         if: steps.wait-for-deploy.outputs.conclusion == 'failure'
+         run: exit 1
+
+       - name: Trigger OWASP Testing
+         uses: benc-uk/workflow-dispatch@v1.1
+         with:
+           workflow: owasp
+           token: ${{ secrets.ACTIONS_API_ACCESS_TOKEN }}
+           inputs: '{"environment": "dev"}'
+
+       - name: Generate Tag from PR Number
+         id: tag_version
+         run: |
+           MATCH="Merge pull request #[0-9]+"
+           STRING=$(curl -s -X GET "https://api.github.com/repos/${{github.repository}}/commits/${{github.sha}}" | jq -r  ".commit.message")
```

Comment:
I think instead of curl, you can use
${{ github.event.push.commits[0].message }}

Reply:
although it might work, I am going to leave it as is now knowing my bit works and add this as a future change